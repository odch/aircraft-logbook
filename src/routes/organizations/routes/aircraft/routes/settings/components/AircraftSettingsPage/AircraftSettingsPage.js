import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../../../../../../../containers/HeaderContainer'
import AircraftSettings from '../../containers/AircraftSettingsContainer'

const AircraftSettingsPage = ({ router: { match } }) => (
  <React.Fragment>
    <Header />
    <AircraftSettings match={match} />
  </React.Fragment>
)

AircraftSettingsPage.propTypes = {
  router: PropTypes.shape({
    match: PropTypes.shape({
      params: PropTypes.object.isRequired
    }).isRequired
  })
}

export default AircraftSettingsPage
