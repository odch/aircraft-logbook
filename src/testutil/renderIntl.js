import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
