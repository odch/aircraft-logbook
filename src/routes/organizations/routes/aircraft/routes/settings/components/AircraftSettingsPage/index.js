import AircraftSettingsPage from './AircraftSettingsPage'

export default AircraftSettingsPage
