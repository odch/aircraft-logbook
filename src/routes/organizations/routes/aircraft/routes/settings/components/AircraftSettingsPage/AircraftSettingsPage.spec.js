import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import renderIntl from '../../../../../../../../testutil/renderIntl'
import AircraftSettingsPage from './AircraftSettingsPage'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('routes', () => {
          describe('settings', () => {
            describe('components', () => {
              describe('AircraftSettingsPage', () => {
                it('renders correctly', () => {
                  const state = {
                    firebase: {
                      auth: {
                        isEmpty: false,
                        email: 'test@opendigital.ch'
                      },
                      profile: {}
                    },
                    main: {
                      app: {
                        organizations: [{ id: 'my_org' }]
                      }
                    },
                    firestore: {
                      data: {
                        organizationAircrafts: {
                          o7flC7jw8jmkOfWo8oyA: {
                            id: 'o7flC7jw8jmkOfWo8oyA',
                            registration: 'HBKFW'
                          },
                          BKi7HYAIoe1i75H3LMk1: {
                            id: 'BKi7HYAIoe1i75H3LMk1',
                            registration: 'HBKOF'
                          }
                        }
                      }
                    },
                    aircraftSettings: {
                      createCheckDialog: {
                        open: false
                      },
                      deleteCheckDialog: {
                        open: false
                      },
                      createFuelTypeDialog: {
                        open: false
                      }
                    }
                  }

                  const store = configureStore()(state)

                  const props = {
                    router: {
                      match: {
                        params: {
                          organizationId: 'my_org',
                          aircraftId: 'o7flC7jw8jmkOfWo8oyA'
                        }
                      }
                    }
                  }

                  const tree = renderIntl(
                    <Provider store={store}>
                      <Router>
                        <AircraftSettingsPage {...props} />
                      </Router>
                    </Provider>
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })
              })
            })
          })
        })
      })
    })
  })
})
