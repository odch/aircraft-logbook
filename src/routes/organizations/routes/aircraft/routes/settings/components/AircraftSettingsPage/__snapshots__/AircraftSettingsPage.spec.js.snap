// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft routes settings components AircraftSettingsPage renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root Header-grow-2 MuiTypography-h6 MuiTypography-colorInherit"
      >
        <a
          className="Header-homeLink-4"
          href="/"
          onClick={[Function]}
        >
          Vuela
        </a>
      </h6>
      <div>
        <button
          aria-haspopup="true"
          aria-owns={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-cy="user-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </header>,
  <div
    className="injectIntl(AircraftSettings)-container-99"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      HBKFW
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root injectIntl(AircraftSettings)-closeSettingsButton-102 MuiButton-text MuiButton-textPrimary"
        href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Einstellungen schliessen
        </span>
      </a>
    </h4>
    <div>
      <h5
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Checks
      </h5>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Check hinzufügen
        </span>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
      >
        Keine Checks erfasst
      </p>
    </div>
    <div>
      <h5
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Treibstofftypen
      </h5>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Treibstofftyp hinzufügen
        </span>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
      >
        Keine Treibstofftypen erfasst
      </p>
    </div>
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Erweiterte Einstellungen
    </h5>
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-container"
      >
        <div
          className="MuiListItem-root MuiListItem-secondaryAction"
          disabled={false}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Motorstunden-Zähler aktiviert
            </span>
          </div>
        </div>
        <div
          className="MuiListItemSecondaryAction-root"
        >
          <span
            className="MuiSwitch-root MuiSwitch-edgeEnd"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-156 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <span
                  className="MuiSwitch-thumb"
                />
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-159 MuiSwitch-input"
                  onChange={[Function]}
                  type="checkbox"
                />
              </span>
            </span>
            <span
              className="MuiSwitch-track"
            />
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-container"
      >
        <div
          className="MuiListItem-root MuiListItem-secondaryAction"
          disabled={false}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
            >
              Techlog aktiviert
            </span>
          </div>
        </div>
        <div
          className="MuiListItemSecondaryAction-root"
        >
          <span
            className="MuiSwitch-root MuiSwitch-edgeEnd"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-156 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <span
                  className="MuiSwitch-thumb"
                />
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-159 MuiSwitch-input"
                  onChange={[Function]}
                  type="checkbox"
                />
              </span>
            </span>
            <span
              className="MuiSwitch-track"
            />
          </span>
        </div>
      </li>
    </ul>
    <hr
      className="MuiDivider-root injectIntl(AircraftSettings)-divider-100"
    />
    <div
      className="injectIntl(AircraftSettings)-deleteButtonContainer-101"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        data-cy="aircraft-delete-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          Flugzeug löschen
        </span>
      </button>
    </div>
  </div>,
]
`;
